src/integration-builder/initialization.js

```diff
@@ -95,9 +112,168 @@ var initialization = {
+                user = mParticle.Identity.getCurrentUser();
+
+            if (user) {
+                consentState = user.getConsentState();
+
+                if (!consentState) {
+                    consentState = mParticle.Consent.createConsentState();
+                }
+
+                OneTrust.getVendorConsentsRequestV2(function(
+                    oneTrustVendorConsent
+                ) {
+                    // oneTrustVendorConsent object is in the shape of
+                    // {
+                    //     ...
+                    //     addtlConsent:
+                    //         '1~39.43.46.55.61.70.83.89.93.108.117.122.124.131',
+                    //     vendor: {
+                    //         consents: '10100111011'  //
+                    //     },
+                    //     ...
+                    // };
+                    //
+                    // Google Vendors are in addtlConsent - to the right of
+                    // "1~" are ids for which Google Vendor the user has consented,
+                    // ie. 39, 43, 46, etc, but not 1, 2, 3 which are not in this list
+                    //
+                    // IAB Vendors are in vendor.consents
+                    // 1 = consented, 0 = rejected. The index+1 is the IAB Vendor ID
+                    // ie. If IAB ID is 5, the index is 4 in the consent string
+                    // to the left, or 0 (rejected))
+                    setGoogleVendorRequests(
+                        oneTrustVendorConsent,
+                        self.googleVendorConsentMapping,
+                        consentState,
+                        location
+                    );
+
+                    setIABVendorRequests(
+                        oneTrustVendorConsent,
+                        self.iabVendorConsentMapping,
+                        consentState,
+                        location
+                    );
+
+                    setGeneralVendorRequests(
+                        // general vendor consent is on the OptAnonActiveGroups string, similar to GDPR consent
+                        self.getConsentGroupIds(),
+                        self.generalVendorConsentMapping,
+                        consentState,
+                        location
+                    );
+                });
+
+                user.setConsentState(consentState);
+            }
+        }
+    },
 };
 
+function setGoogleVendorRequests(
+    oneTrustVendorConsent,
+    googleVendorConsentMapping,
+    consentState,
+    location
+) {
+    try {
+        // turn Google Vendor Consent string into an array of IDs
+        var googleConsentedVendors = oneTrustVendorConsent
+            ? oneTrustVendorConsent.addtlConsent
+                  .slice()
+                  .split('~')[1]
+                  .split('.')
+            : null;
+
+        if (googleConsentedVendors && googleConsentedVendors.length) {
+            for (var vendorId in googleVendorConsentMapping) {
```

Comment:
This might be a nitpick, but these three `set-NAME-VendorRequests` share a lot of common functionality. Is there a value in refactoring this loop into a shared function to keep things DRY?

Reply:
While there is a for loop in each vendor consent method, each one actually has different code within the for block.  So I think rather than passing a function with the different code into a function with generic for loop, I went for readability because the OneTrust vendor consent API is already unintuitive.

However, I am open to an implementation if it's easy to read and simplifies the code.

Reply:
I think readability is more important than keeping things DRY in this case, since there are only three instances of this similar functionality.